pkg/restapi/restapi.go

```diff
@@ -86,6 +86,26 @@ func ErrorHandler() func(error, echo.Context) {
 	}
 }
 
+func GetAcceptHeaderContentType(c echo.Context, supportedContentTypes ...string) (string, error) {
+	ctype := c.Request().Header.Get(echo.HeaderAccept)
+	for _, supportedContentType := range supportedContentTypes {
+		if strings.HasPrefix(ctype, supportedContentType) {
+			return supportedContentType, nil
+		}
+	}
+	return "", echo.ErrUnsupportedMediaType
```

Comment:
this should return a new error
`ErrNotAcceptable = echo.NewHTTPError(http.StatusNotAcceptable)`